src/SFA.DAS.AssessorService.Application/Handlers/OrganisationStandards/OrganisationStandardVersionOptInHandler.cs

```diff
@@ -0,0 +1,108 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AssessorService.Api.Types.Consts;
+using SFA.DAS.AssessorService.Api.Types.Models;
+using SFA.DAS.AssessorService.Api.Types.Models.AO;
+using SFA.DAS.AssessorService.Application.Interfaces;
+using SFA.DAS.AssessorService.ApplyTypes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.Apply
+{
+    public class OrganisationStandardVersionOptInHandler : IRequestHandler<OrganisationStandardVersionOptInRequest, OrganisationStandardVersion>
+    {
+        private readonly IOrgansiationStandardRepository _repository;
+        private readonly IApplyRepository _applyRepository;
+        private readonly IStandardRepository _standardRepository;
+        private readonly IContactQueryRepository _contactQueryRepository;
+        private readonly IEMailTemplateQueryRepository _eMailTemplateQueryRepository;
+        private readonly IMediator _mediator;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly ILogger<OrganisationStandardVersionOptInHandler> _logger;
+
+        public OrganisationStandardVersionOptInHandler(IOrgansiationStandardRepository repository, IApplyRepository applyRepository, IStandardRepository standardRepository,
+            IContactQueryRepository contactQueryRepository, IEMailTemplateQueryRepository eMailTemplateQueryRepository, IMediator mediator,
+            IUnitOfWork unitOfWork, ILogger<OrganisationStandardVersionOptInHandler> logger)
+        {
+            _repository = repository;
+            _applyRepository = applyRepository;
+            _standardRepository = standardRepository;
+            _contactQueryRepository = contactQueryRepository;
+            _eMailTemplateQueryRepository = eMailTemplateQueryRepository;
+            _mediator = mediator;
+            _unitOfWork = unitOfWork;
+            _logger = logger;
+        }
+
+        public async Task<OrganisationStandardVersion> Handle(OrganisationStandardVersionOptInRequest request, CancellationToken cancellationToken)
+        {
+            try
+            {
+                _unitOfWork.Begin();
+
+                OrganisationStandardVersion orgStandardVersion = null;
+
+                var orgStandard = await _repository.GetOrganisationStandardByOrganisationIdAndStandardReference(request.EndPointAssessorOrganisationId, request.StandardReference);
+                if (orgStandard != null)
```

Comment:
Does this not need to error in some way if they are not yet approved to assess any version of the standard? Instead of returning null

Reply:
If they aren't approved for the standard, the opt in should fail ( granted shouldn't hit this scenario but defensive coding and all that ) also what if the version already exists?

Reply:
Both scenarios are covered earlier in the flow. But i've added an additional check